src/recipe.h

```diff
@@ -50,6 +51,26 @@ struct recipe_proficiency {
     void deserialize( JsonIn &jsin );
 };
 
+struct book_recipe_data {
+    int skill_req = -1;
+    bool use_alt_name = false;
+    std::string alt_name = "";
+    bool hidden = false;
+
+    book_recipe_data( JsonArray arr ) {
```

Comment:
Please move implementation to the .cpp file. Also, the convention is to read using a `deserialize` function.